src/renderer/events/Navigation.js

```diff
@@ -0,0 +1,1193 @@
+/**
+ * Copyright (c) 2017-present, Raphael Amorim.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/*
+  Fork from luke-chang/js-spatial-navigation
+ */
+
+let GlobalConfig = {
+  selector: '',
+  straightOnly: false,
+  straightOverlapThreshold: 0.5,
+  rememberSource: false,
+  disabled: false,
+  defaultElement: '',
+  enterTo: '',
+  leaveFor: null,
+  restrict: 'self-first',
+  tabIndexIgnoreList:
+    'a, input, select, textarea, button, iframe, [contentEditable=true]',
+  navigableFilter: null,
+};
+
+const KEYMAPPING = {
+  '37': 'left',
+  '38': 'up',
+  '39': 'right',
+  '40': 'down',
+};
+
+const REVERSE = {
+  left: 'right',
+  up: 'down',
+  right: 'left',
+  down: 'up',
+};
+
+const EVENT_PREFIX = 'sn:';
+const ID_POOL_PREFIX = 'section-';
+
+let _idPool = 0;
+let _ready = false;
+let _pause = false;
+let _sections = {};
+let _sectionCount = 0;
+let _defaultSectionId = '';
+let _lastSectionId = '';
+let _duringFocusChange = false;
+
+var elementMatchesSelector =
+  Element.prototype.matches ||
+  Element.prototype.matchesSelector ||
+  Element.prototype.mozMatchesSelector ||
+  Element.prototype.webkitMatchesSelector ||
+  Element.prototype.msMatchesSelector ||
+  Element.prototype.oMatchesSelector ||
+  function(selector) {
+    var matchedNodes = (this.parentNode || this.document).querySelectorAll(
+      selector
+    );
+    return [].slice.call(matchedNodes).indexOf(this) >= 0;
+  };
+
+function getRect(elem) {
+  var cr = elem.getBoundingClientRect();
+  var rect = {
+    left: cr.left,
+    top: cr.top,
+    right: cr.right,
+    bottom: cr.bottom,
+    width: cr.width,
+    height: cr.height,
+  };
+  rect.element = elem;
+  rect.center = {
+    x: rect.left + Math.floor(rect.width / 2),
+    y: rect.top + Math.floor(rect.height / 2),
+  };
+  rect.center.left = rect.center.right = rect.center.x;
+  rect.center.top = rect.center.bottom = rect.center.y;
+  return rect;
+}
+
+function partition(rects, targetRect, straightOverlapThreshold) {
+  var groups = [[], [], [], [], [], [], [], [], []];
+
+  for (var i = 0; i < rects.length; i++) {
```

Comment:
What do you think of making the following change from line 90 to line 94?
It's just a basic refactoring
```js
  for (const rect of rects) {
    const center = rect.center;
    let x;
    let y;
    let groupId;
```

Reply:
This code is borrowed from github.com/luke-chang/js-spatial-navigation
Before refactor it, I need to write some tests